apps/backend/src/modules/quiz-answers/quiz-answer.service.ts

```diff
@@ -0,0 +1,138 @@
+import { ErrorMessage } from "~/libs/enums/enums.js";
+import { type Service } from "~/libs/types/types.js";
+
+import { type CategoryService } from "../categories/categories.js";
+import { INITIAL_SCORE } from "./libs/constants/constants.js";
+import { HTTPCode } from "./libs/enums/enums.js";
+import { QuizError } from "./libs/exceptions/exceptions.js";
+import {
+	type QuizAnswer,
+	type QuizAnswersResponseDto,
+	type QuizScoreDto,
+} from "./libs/types/types.js";
+import { QuizAnswerEntity } from "./quiz-answer.entity.js";
+import { type QuizAnswerRepository } from "./quiz-answer.repository.js";
+
+class QuizAnswerService implements Service {
+	private categoryService: CategoryService;
+
+	private quizAnswerRepository: QuizAnswerRepository;
+
+	public constructor(
+		quizAnswerRepository: QuizAnswerRepository,
+		categoryService: CategoryService,
+	) {
+		this.quizAnswerRepository = quizAnswerRepository;
+		this.categoryService = categoryService;
+	}
+
+	public async create(payload: QuizAnswer): Promise<QuizAnswer> {
+		const item = await this.quizAnswerRepository.create(
+			QuizAnswerEntity.initializeNew(payload),
+		);
+
+		return item.toObject();
+	}
+
+	public async createScores({
+		answerIds,
+		userId,
+	}: {
+		answerIds: number[];
+		userId: number;
+	}): Promise<QuizScoreDto[]> {
+		const countByCategoryId = new Map<number, number>();
+		const scores = [];
```

Comment:
Add type here

Reply:
Not resolved

Reply:
Thought of parameters insead of return type. Fixed now ✅